### PR TITLE
Refactors builder.toml generation action

### PR DESCRIPTION
**Pull request by ryanmoran**
- uses more functional patterns to increase testability
- replaces custom image reference parsing with external library

### CODE REVIEW

File: actions/builder/update/entrypoint/' [hunk 1]
```diff
@@ -0,0 +1,311 @@
+package main_test
```
Comment (sophiewigmore) [inline actions/builder/update/entrypoint/':1]: I'm a bit confused about the name of this file being `'` is there a reason for this or is it a typo?

File: actions/builder/update/entrypoint/main.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	if options.BuilderFile == "" {
+		fail(errors.New(`missing required input "builder-file"`))
+	}
+
+	if options.RegistryServer == "" {
+		fail(errors.New(`missing required input "registry-server"`))
+	}
+
+	builder, err := internal.ParseBuilderFile(options.BuilderFile)
```
Comment (sophiewigmore) [inline actions/builder/update/entrypoint/main.go:32]: Will this work the first time it runs when there is no builder.toml file yet?

Comment (ryanmoran) [reply actions/builder/update/entrypoint/main.go:32]: The assumption is that the `builder.toml` is checked in from the very beginning. The first time, a human will have created a `builder.toml` by hand.

File: actions/builder/update/entrypoint/' (hunk 1 above)
Comment (ryanmoran) [reply actions/builder/update/entrypoint/':1]: Yeah. Looks like this got checked in accidentally.

File: actions/builder/update/entrypoint/main.go (hunk 2 above)
Comment (sophiewigmore) [reply actions/builder/update/entrypoint/main.go:32]: Sounds good, I didn't realize that was the assumption

**Review by sophiewigmore — APPROVED** [review]

**Pull request merged**